enterprise/users/src/test/java/io/crate/integrationtests/UserManagementIntegrationTest.java

```diff
@@ -139,24 +142,40 @@ public void testDropUserIfExists() throws Exception {
     @Test
     public void testDropUserUnAuthorized() throws Exception {
         expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("User \"normal\" is not authorized to execute the statement");
+        expectedException.expectMessage("Missing 'AL' privilege for user 'normal'");
         executeAsNormalUser("drop user ford");
     }
 
     @Test
     public void testCreateUserUnAuthorized() throws Exception {
         expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("User \"normal\" is not authorized to execute the statement");
+        expectedException.expectMessage("Missing 'AL' privilege for user 'normal'");
         executeAsNormalUser("create user ford");
     }
 
     @Test
     public void testCreateNormalUserUnAuthorized() throws Exception {
         expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("User \"normal\" is not authorized to execute the statement");
+        expectedException.expectMessage("Missing 'AL' privilege for user 'normal'");
         executeAsNormalUser("create user ford");
     }
 
+    @Test
+    public void test_create_user_as_regular_user_with_al_privileges() throws Exception {
+        executeAsSuperuser("CREATE USER trillian");
+        executeAsSuperuser("GRANT AL TO trillian");
+        assertBusy(() -> {
```

Comment:
why do we need an `assertBusy` here? why not just use `assertUserIsCreated` or nothing as the next statement will fail otherwise anyway?

Reply:
I added it to help with debugging. Initially I used the `executeAsNormalUser` which didn't work - I thought it was due to the `GRANT AL` not working, but it turned out to be the case because the session is created eagerly and isn't reloaded on privilege changes. 

The `assertBusy` probably wouldn't be required. I wasn't sure if the cluster state changes are fully propagated to the user manager already once the `GRANT` statement returns.

Reply:
Isn't the new `executeAs(...)` solving the issue you had with `executeAsNormalUser`? If so I still think, the whole block can be removed to make the test scenario is smaller.